core/src/com/unciv/ui/pickerscreens/TechPickerScreen.kt

```diff
@@ -17,7 +17,7 @@ import java.util.*
 import kotlin.collections.ArrayList
 
 
-class TechPickerScreen(internal val civInfo: CivilizationInfo, centerOnTech: Technology? = null) : PickerScreen() {
+class TechPickerScreen(internal val civInfo: CivilizationInfo, centerOnTech: Technology? = null, freeTechPick: Boolean = false) : PickerScreen() {
 
     private var techNameToButton = HashMap<String, TechButton>()
     private var isFreeTechPick: Boolean = false
```

Comment:
This is never changed... so why even have this var, if its only value is that of freeTechPick?

Reply:
How should it be implemented? As far as my limited knowledge of Kotlin goes, this is the only way to pass a parameter to the init {} block? If I make it a `private val`, Android Studio starts acting up that it is never used outside of the init {} block.

Reply:
If I prefix `private val` on `freeTechPick`, drop isFreeTechPick and replace its uses with the former -> no compiler or inspection complaints. Just the way old name shadowed and until ones... Runs fine, too. AS/Plugin version diff maybe.

So that would be a class-level private val initialized automatically with the same-named constructor parameter. Accessibility in the anonymous function created in init{} not a problem. However, a simple constructor parameter without a backing field should work just as well, as the anon would get it via closure? Nope, `setButtonsInfo` and `selectTechnology` want it - so - first alternative.

Reply:
Ah thanks, that solved it. Cleaning my branches is something I maybe should do sometime